### DISCUSSION TITLE
Support for TimescaleDB

### DISCUSSION THREAD

**Discussion by theelderbeever** [piccolo-orm/piccolo#442] [Ideas]
It would be really cool to see support for some the Postgres wrappers out there. One such example is [TimescaleDB](https://www.timescale.com/) which adds columnar storage to Postgres. Right now `piccolo` pretty much works out of the box for generating the basic table structure and then the extras like hyptertables can be done via `.raw` migrations.

That said if it ever feels doable it would be really cool to support some of their extra functions and utilities in python to heavily support python data applications.

Async python, async ORM, plus analytics engine with columnar storage would be the bees knees!

**Comment by dantownsend** [comment]
@theelderbeever I agree. I used Timescale in work for a while, and it seems to be getting more and more traction.

Even if there was some kind of extension for creating Hypertables.

I think that's why Postgres is so interesting at the moment, with all of the cool extensions.

**Reply by dantownsend** [reply]
Just getting some thoughts down for future reference.

The API could look something like:

```python
# piccolo/extensions/timescaledb.py

class Hypertable(Table):
    def create_table(cls):
        # overrides create table, so it also creates a hypertable
        ...

```

And then

```python
class MyTimeseriesTable(Hypertable):
    ...
```
